src/services/resource.js

```diff
@@ -226,13 +234,115 @@ module.exports = class resourceHelper {
 		// generic function to merge all the collected data about the resource
 		result = await this.responseBuilder(response, userDetails, orgDetails, additionalResourceInformation)
 		// count of requested for changes resources
-		result.changes_requested_count = distinctInreviewResourceIds.count > 0 ? distinctInreviewResourceIds.count : 0
+		result.changes_requested_count = requestedForChangesResources.count > 0 ? requestedForChangesResources.count : 0
 		return responses.successResponse({
 			statusCode: httpStatusCode.ok,
 			message: 'RESOURCE_LISTED_SUCCESSFULLY',
 			result,
 		})
 	}
+
+	/**
+	 * Return the mapping of resource Id and its final status
+	 * @name finalResourceStatus
+	 * @param {Array} resourceId - Array of unique resource ids
+	 * @param {Array} organizationId - Array of unique org ids
+	 * @returns {JSON} - List of resources with final status
+	 */
+	static async finalResourceStatus(resourceId, organizationId) {
+		let recourceStatusMapping = {}
+
+		// Get the review details of all the resources created by the logged-in user
+		const resourceReviews = await reviewsQueries.findAll(
+			{
+				organization_id: {
+					[Op.in]: organizationId,
+				},
+				resource_id: {
+					[Op.in]: resourceId,
+				},
+			},
+			['resource_id', 'status']
+		)
+
+		if (resourceReviews.length > 0) {
+			// Create a mapping of resource id and review statuses
+			resourceReviews.forEach((review) => {
+				if (!recourceStatusMapping[review.resource_id]) {
+					recourceStatusMapping[review.resource_id] = []
+				}
+				// Push only unique status values to the object
+				if (!recourceStatusMapping[review.resource_id].includes(review.status)) {
+					recourceStatusMapping[review.resource_id].push(review.status)
+				}
+			})
+
+			recourceStatusMapping = await this.determineResourceStatus(recourceStatusMapping)
+		}
+
+		return recourceStatusMapping
+	}
+
+	/**
+	 * Determine the final status of each resources - logic
+	 * @name determineResourceStatus
+	 * @param {Object} reviewDetails - Object of each review detail
+	 * @returns {JSON} - Final determined status of each reviews
+	 */
+	static async determineResourceStatus(reviewDetails) {
```

Comment:
@priyanka-TL just change this function to take array of status and give a string of status

Reply:
@aks30 Updated to return a single string from array of status.